src/app.js

```diff
@@ -2,0 +1,49 @@
+/* eslint-disable no-console */
+const fs = require('fs');
+const path = require('path');
+
+const moveFile = (targetFile, moveTo) => {
+  try {
+    if (targetFile === moveTo) {
+      console.log('Initial file and target destination are the same');
+
+      return;
+    }
+
+    let targetIsDirectory = false;
+
+    if (fs.existsSync(moveTo)) {
+      targetIsDirectory = fs.lstatSync(moveTo).isDirectory();
+    } else if (moveTo.endsWith('/')) {
+      console.error('Error: Destination directory does not exist');
+
+      return;
+    }
+
+    const destinationPath = targetIsDirectory
+      ? path.join(moveTo, path.basename(targetFile))
+      : moveTo;
+
+    fs.rename(targetFile, destinationPath, (err) => {
+      if (err) {
+        console.error('Error happened during the rename operation:', err);
```

Comment:
It's good to provide detailed error messages, but consider adding more context to the error message to help users understand what might have gone wrong.